common/src/main/java/org/apache/falcon/entity/parser/ClusterEntityParser.java

```diff
@@ -243,58 +244,95 @@ private void validateLocations(Cluster cluster) throws ValidationException {
         try {
             fs = HadoopClientFactory.get().createProxiedFileSystem(conf);
         } catch (FalconException e) {
-            throw new ValidationException(
-                    "Unable to get file system handle for cluster " + cluster.getName(), e);
+            throw new ValidationException("Unable to get file system handle for cluster " + cluster.getName(), e);
         }
 
-        for (Location location : cluster.getLocations().getLocations()) {
-            final String locationName = location.getName();
-            if (locationName.equals("temp")) {
-                continue;
-            }
+        Location stagingLocation = ClusterHelper.getLocation(cluster, ClusterLocationType.STAGING);
 
-            try {
-                checkPathOwnerAndPermission(cluster.getName(), location.getPath(), fs,
-                        "staging".equals(locationName)
-                                ? HadoopClientFactory.ALL_PERMISSION
-                                : HadoopClientFactory.READ_EXECUTE_PERMISSION);
-            } catch (IOException e) {
-                throw new ValidationException("Unable to validate the location of name: "
-                        + location.getName() + " with path:" + location.getPath()
-                        + " for cluster " + cluster.getName(), e);
+        if (stagingLocation == null) {
```

Comment:
From the older version of the code, it seems like staging perms need to be ALL and data perms need to be Read/Execute. But this seems to be changing the permission scheme. Can you confirm ?